File: app/src/main/java/com/unid/myapplication/WelcomeActivity.java
package com.unid.myapplication;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class WelcomeActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_welcome);

        TextView userName = (TextView) findViewById(R.id.username);

        Intent beforeIntent = getIntent();
        String username = beforeIntent.getStringExtra("name");
        userName.setText(username);

        Button startBtn = (Button) findViewById(R.id.startBtn);
        startBtn.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View view) {
                Intent intent = new Intent(WelcomeActivity.this, StartActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/com/unid/myapplication/WelcomeActivity.kt
package com.unid.myapplication

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class WelcomeActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_welcome)
        val userName = findViewById<View>(R.id.username) as TextView
        val beforeIntent = intent
        val username = beforeIntent.getStringExtra("name")
        userName.text = username
        val startBtn = findViewById<View>(R.id.startBtn) as Button
        startBtn.setOnClickListener {
            val intent = Intent(this@WelcomeActivity, StartActivity::class.java)
            startActivity(intent)
        }
    }
}